<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN" "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<title>TestNG:  Unit Test</title>
<style type="text/css">
table caption,table.info_table,table.result,table.passed,table.failed {margin-bottom:10px;border:1px solid #000099;border-collapse:collapse;empty-cells:show;}
table.info_table td,table.info_table th,table.result td,table.result th,table.passed td,table.passed th,table.failed td,table.failed th {
border:1px solid #000099;padding:.25em .5em .25em .5em
}
table.result th {vertical-align:bottom}
tr.param th {padding-left:1em;padding-right:1em}
tr.param td {padding-left:.5em;padding-right:2em}
td.numi,th.numi,td.numi_attn {
text-align:right
}
tr.total td {font-weight:bold}
table caption {
text-align:center;font-weight:bold;
}
table.passed tr.stripe td,table tr.passedodd td {background-color: #00AA00;}
table.passed td,table tr.passedeven td {background-color: #33FF33;}
table.passed tr.stripe td,table tr.skippedodd td {background-color: #cccccc;}
table.passed td,table tr.skippedodd td {background-color: #dddddd;}
table.failed tr.stripe td,table tr.failedodd td,table.result td.numi_attn {background-color: #FF3333;}
table.failed td,table tr.failedeven td,table.result tr.stripe td.numi_attn {background-color: #DD0000;}
tr.stripe td,tr.stripe th {background-color: #E6EBF9;}
p.totop {font-size:85%;text-align:center;border-bottom:2px black solid}
div.shootout {padding:2em;border:3px #4854A8 solid}
</style>
</head>
<body>
<table cellspacing="0" cellpadding="0" class="result">
<tr><th>Test</th><th class="numi">Methods<br/>Passed</th><th class="numi">Scenarios<br/>Passed</th><th class="numi"># skipped</th><th class="numi"># failed</th><th class="numi">Total<br/>Time</th><th class="numi">Included<br/>Groups</th><th class="numi">Excluded<br/>Groups</th></tr>
<tr><td style="text-align:left;padding-right:2em">test4Demo</td><td class="numi">6</td><td class="numi">5</td><td class="numi">0</td><td class="numi_attn">4</td><td class="numi">1.7 seconds</td><td class="numi"></td><td class="numi"></td></tr>
</table>
<table cellspacing="0" cellpadding="0" class="passed" id="summary">
<tr><th>Class</th><th>Method</th><th># of<br/>Scenarios</th><th>Start</th><th>Time<br/>(ms)</th></tr>
<tr><th colspan="4">test4Demo &#8212; failed</th></tr>
<tr class="failedodd"><td rowspan="4">sence.loginTest</td><td><a href="#m1"><b>runScenario</b> (login) </a></td><td class="numi">1</td><td>1566181903063</td><td class="numi">27</td></tr><tr class="failedodd"><td><a href="#m2"><b>runScenario</b> (login) </a></td><td class="numi">1</td><td>1566181903063</td><td class="numi">27</td></tr><tr class="failedodd"><td><a href="#m3"><b>runScenario</b> (login) </a></td><td class="numi">1</td><td>1566181903063</td><td class="numi">27</td></tr><tr class="failedodd"><td><a href="#m4"><b>runScenario</b> (login) </a></td><td class="numi">1</td><td>1566181903063</td><td class="numi">27</td></tr>
<tr><th colspan="4">test4Demo &#8212; passed</th></tr>
<tr class="passedodd"><td rowspan="2">sence.createServerTest</td><td><a href="#m5"><b>runScenario</b> (createServer) ("鏂板鏈嶅姟")</a></td><td class="numi">2</td><td>1566181901616</td><td class="numi">1392</td></tr><tr class="passedodd"><td><a href="#m6"><b>runScenario</b> (createServer) ("鏂板鏈嶅姟")</a></td><td class="numi">2</td><td>1566181901616</td><td class="numi">1392</td></tr>
<tr class="passedeven"><td rowspan="4">sence.loginTest</td><td><a href="#m7"><b>runScenario</b> (login) </a></td><td class="numi">3</td><td>1566181903021</td><td class="numi">40</td></tr><tr class="passedeven"><td><a href="#m8"><b>runScenario</b> (login) </a></td><td class="numi">3</td><td>1566181903021</td><td class="numi">40</td></tr><tr class="passedeven"><td><a href="#m9"><b>runScenario</b> (login) </a></td><td class="numi">3</td><td>1566181903021</td><td class="numi">40</td></tr><tr class="passedeven"><td><a href="#m10"><b>runScenario</b> (login) </a></td><td class="numi">3</td><td>1566181903021</td><td class="numi">40</td></tr>
</table>
<h1>test4Demo</h1>
<h2 id="m1">sence.loginTest:runScenario</h2>
<table cellspacing="0" cellpadding="0" class="result">
<tr class="param"><th>Parameter #1</th></tr>
<tr class="param stripe"><td>04 - case5:输入错误密码123</td>
</tr>
<tr><td style="padding-left:3em" colspan="1">
<h3>Test Messages</h3>
=============鎵ц缁撴灉: Failed=============<br/>
<h3>Failure</h3>
<p>04 - case5:输入错误密码123
 throw an exception:
java.lang.RuntimeException: Invoke class (sence.loginTest.java:1) method[checkMsg] error:
Invoke spec method[checkMsg] error:java.lang.AssertionError: expected [用户名不存在！] but found [密码不正确！]!
java.lang.RuntimeException: Invoke class (sence.loginTest.java:1) method[checkMsg] error:
Invoke spec method[checkMsg] error:java.lang.AssertionError: expected [用户名不存在！] but found [密码不正确！]!
	at org.test4j.spec.JSpecExecutorFactory.runSingleStep(JSpecExecutorFactory.java:216)
	at org.test4j.spec.JSpecExecutorFactory.runScenario(JSpecExecutorFactory.java:178)
	at org.test4j.testng.JSpec.run(JSpec.java:75)
	at sence.loginTest.runScenario(loginTest.java:23)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:80)
	at org.testng.internal.MethodInvocationHelper$1.runTestMethod(MethodInvocationHelper.java:182)
	at org.test4j.testng.Test4JHookable.run(Test4JHookable.java:56)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeHookable(MethodInvocationHelper.java:194)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:707)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:901)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1231)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:127)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:111)
	at org.testng.TestRunner.privateRun(TestRunner.java:767)
	at org.testng.TestRunner.run(TestRunner.java:617)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:334)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:329)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:291)
	at org.testng.SuiteRunner.run(SuiteRunner.java:240)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:86)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1198)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1123)
	at org.testng.TestNG.run(TestNG.java:1031)
	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:293)
	at org.apache.maven.surefire.testng.TestNGXmlTestSuite.execute(TestNGXmlTestSuite.java:84)
	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:91)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:200)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:153)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
Caused by: java.lang.RuntimeException: Invoke spec method[checkMsg] error:java.lang.AssertionError: expected [用户名不存在！] but found [密码不正确！]
	at org.test4j.spec.scenario.step.SpecMethod.execute(SpecMethod.java:98)
	at org.test4j.spec.JSpecExecutorFactory.runSingleStep(JSpecExecutorFactory.java:210)
	... 37 more
Caused by: java.lang.RuntimeException: java.lang.AssertionError: expected [用户名不存在！] but found [密码不正确！]
	at org.test4j.tools.commons.ExceptionWrapper.wrapWithRuntimeException(ExceptionWrapper.java:106)
	at org.test4j.tools.reflector.MethodAccessor.invokeUnThrow(MethodAccessor.java:75)
	at org.test4j.spec.scenario.step.SpecMethod.execute(SpecMethod.java:93)
	... 38 more
Caused by: java.lang.AssertionError: expected [用户名不存在！] but found [密码不正确！]
	at org.testng.Assert.fail(Assert.java:94)
	at org.testng.Assert.failNotEquals(Assert.java:494)
	at org.testng.Assert.assertEquals(Assert.java:123)
	at org.testng.Assert.assertEquals(Assert.java:176)
	at org.testng.Assert.assertEquals(Assert.java:186)
	at sence.loginTest.checkMsg(loginTest.java:55)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.test4j.tools.reflector.MethodAccessor.invoke(MethodAccessor.java:49)
	at org.test4j.tools.reflector.MethodAccessor.invokeUnThrow(MethodAccessor.java:72)
	... 39 more

04 - case5:输入错误密码123 throw errors at:</p>
org.test4j.spec.JSpecExecutorFactory.runScenario(JSpecExecutorFactory.java:191)
<br/>at org.test4j.testng.JSpec.run(JSpec.java:75)
<br/>at sence.loginTest.runScenario(loginTest.java:23)
<br/>12 lines not shown
</td></tr>
</table>
<p class="totop"><a href="#summary">back to summary</a></p>
<h2 id="m2">sence.createServerTest:runScenario</h2>
<table cellspacing="0" cellpadding="0" class="result">
<tr class="param"><th>Parameter #1</th></tr>
<tr class="param stripe"><td>01 - case1:登录后新增服务，状态=空闲</td>
</tr>
<tr><td style="padding-left:3em" colspan="1">
=============鎵ц缁撴灉: Success=============<br/>
</td></tr>
</table>
<p class="totop"><a href="#summary">back to summary</a></p>
<h2 id="m3">sence.loginTest:runScenario</h2>
<table cellspacing="0" cellpadding="0" class="result">
<tr class="param"><th>Parameter #1</th></tr>
<tr class="param stripe"><td>02 - case2:输入错误用户名</td>
</tr>
<tr><td style="padding-left:3em" colspan="1">
=============鎵ц缁撴灉: Success=============<br/>
</td></tr>
</table>
<p class="totop"><a href="#summary">back to summary</a></p>
<h2 id="m4">sence.loginTest:runScenario</h2>
<table cellspacing="0" cellpadding="0" class="result">
<tr class="param"><th>Parameter #1</th></tr>
<tr class="param stripe"><td>01 - case1:输入正确用户名、密码</td>
</tr>
<tr><td style="padding-left:3em" colspan="1">
=============鎵ц缁撴灉: Success=============<br/>
</td></tr>
</table>
<p class="totop"><a href="#summary">back to summary</a></p>
<h2 id="m5">sence.loginTest:runScenario</h2>
<table cellspacing="0" cellpadding="0" class="result">
<tr class="param"><th>Parameter #1</th></tr>
<tr class="param stripe"><td>03 - case3:输入错误密码</td>
</tr>
<tr><td style="padding-left:3em" colspan="1">
=============鎵ц缁撴灉: Success=============<br/>
</td></tr>
</table>
<p class="totop"><a href="#summary">back to summary</a></p>
<h2 id="m6">sence.createServerTest:runScenario</h2>
<table cellspacing="0" cellpadding="0" class="result">
<tr class="param"><th>Parameter #1</th></tr>
<tr class="param stripe"><td>02 - case2:登录后新增服务，状态=使用中</td>
</tr>
<tr><td style="padding-left:3em" colspan="1">
=============鎵ц缁撴灉: Success=============<br/>
</td></tr>
</table>
<p class="totop"><a href="#summary">back to summary</a></p>
</body></html>
